db.go

```diff
@@ -36,12 +36,81 @@ const (
 // All data access is performed through transactions which can be obtained through the DB.
 // All the functions on DB will return a ErrDatabaseNotOpen if accessed before Open() is called.
 type DB struct {
+	pagePool sync.Pool
+
+	logger Logger
+
+	openFile func(string, int, os.FileMode) (*os.File, error)
+	file     *os.File
+	data     *[maxMapSize]byte
+	meta0    *common.Meta
+	meta1    *common.Meta
+	rwtx     *Tx
+
+	freelist *freelist
+	batch    *batch
+
+	ops struct {
+		writeAt func(b []byte, off int64) (n int, err error)
+	}
+
+	// FreelistType sets the backend freelist type. There are two options. Array which is simple but endures
+	// dramatic performance degradation if database is large and fragmentation in freelist is common.
+	// The alternative one is using hashmap, it is faster in almost all circumstances
+	// but it doesn't guarantee that it offers the smallest page id available. In normal case it is safe.
+	// The default type is array
+	FreelistType FreelistType
+
+	path string
+	// `dataref` isn't used at all on Windows, and the golangci-lint
+	// always fails on Windows platform.
+	//nolint
+	dataref []byte // mmap'ed readonly, write throws SEGV
+	txs     []*Tx
+
 	// Put `stats` at the first field to ensure it's 64-bit aligned. Note that
```

Comment:
please read this comment on the alignment issues we had with some arch (ARM?).

Reply:
Ah good catch, thanks!


``` 
bbolt/db.go:38:9: struct with 336 pointer bytes could be 192
```

I guess still an improvement over 528 bytes